infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/provisioner/labels/SinglePortLabelsProvisioner.java

```diff
@@ -91,13 +91,18 @@ public void provision(DockerEnvironment internalEnv, RuntimeIdentity identity)
   /**
    * Constructs unique traefik service name - contains server, machine names and workspace ID. Dots
    * is not allowed and replaced by dashes. Result is like:
-   * exec-agent-http-dev-machine-workspaceao6k83hkdav975g5
+   * exec-agent-http_dev-machine_workspaceao6k83hkdav975g5
    */
   private String getServiceName(String host) {
     int idx =
         (externalIpOfContainers != null && host.contains(externalIpOfContainers))
             ? host.indexOf(externalIpOfContainers)
             : host.indexOf(internalIpOfContainers);
-    return host.substring(0, idx - 1).replaceAll("\\.", "-");
+    if (idx > 1) {
+      return host.substring(0, idx - 1).replaceAll("\\.", "-");
```

Comment:
Please, review logic here again. We dont need to replace dots anymore. Only cut until first dot and throw out any optional IP  and wildcard domain.

Reply:
good point!